### DISCUSSION TITLE
ccache core-dumps with invalid remote storage URLs

### DISCUSSION THREAD

**Discussion by pushkarnk** [ccache/ccache#1390] [Development]
I thought of starting a discussion on this because I am not fully sure if this behavior qualifies for a bug report.

This is related to a ccache[ bug report in Ubuntu](https://bugs.launchpad.net/ubuntu/+source/ccache/+bug/2051695). With 4.9.1, I notice the remote_url.bash suite passing, albeit with the last two tests core-dumping. They are passed invalid URLs for CCACHE_REMOTE_STORAGE, causing the raising of a `Url::build_error` which is never caught and ends up SIGABRTing cache.

To replicate this behavior of remote_url.bash in a smaller test case:
```
$ // compile the latest master branch of github.com/ccache/ccache
$ cmake -DCMAKE_BUILD_TYPE=Release
$ make install
$ export CCACHE_REMOTE_STORAGE="http://foo@"
$ // write a small c program in test.c
$ ccache gcc -c test.c
terminate called after throwing an instance of 'Url::build_error'
  what():  User info defined, but host is empty
Aborted (core dumped)
```
Is not catching `Url::build_error` and letting the `ccache` utility fail intentional? I gather that catching it and handling it all throughout the code might be an overkill, but I may be wrong here. Can we have a more graceful exit? I am seeking opinions here.

Thank you!

**Comment by jrosdahl** [comment]
It's a bug. Will look into it. Thanks for the notice!

**Reply by jrosdahl** [reply]
Bug report: #1394

**Reply by jrosdahl** [reply]
> With 4.9.1

@pushkarnk: Also note that there is no ccache 4.9.1 yet.

**Reply by pushkarnk** [reply]
Yeah, I misread the Debian packaging release 4.9-1 as upstream release 4.9.1 :)
